File: java/tebak.java
import java.util.LinkedList;
import java.util.Random;
import java.util.Scanner;

public class tebak {
    public static void main(String[] args) {
        LinkedList<Integer> histori = new LinkedList<>();
        String[] data = new String[100];
        Scanner input = new Scanner(System.in);
        Random rand = new Random();
        int index = 0;
        int tebakan;
        int salah = 0;
        String response;

        System.out.println("----------Permainan Tebak Angka----------");
        System.out.println("Tebaklah sebuah angka antara 0 hingga 100");

        do {
            System.out.print("\nMasukkan nama kamu: ");
            data[index] = input.nextLine();
            int randomNum = rand.nextInt(101);
            index++;

            while (true) {
                System.out.print("Masukkan tebakan Anda: ");
                tebakan = input.nextInt();
                input.nextLine();
                if (tebakan < 0 || tebakan > 100) {
                    System.out.println("Tebakan diluar dari angka yang ditentukan. Coba lagi.");
                    continue;
                }
                histori.add(tebakan);
                if (tebakan > randomNum) {
                    System.out.println("Tebakanmu terlalu tinggi. Coba lagi.");
                    salah++;
                    continue;
                } else if (tebakan < randomNum) {
                    System.out.println("Tebakanmu terlalu rendah. Coba lagi.");
                    salah++;
                    continue;
                } else if (tebakan == randomNum) {
                    System.out.println();
                    System.out.println("Selamat! Kamu berhasil menebak angka yang benar setelah " + salah + " kali percobaan");

                    if (salah > 10) {
                        System.out.println("Skill issue lu awokwkwk");
                    }
                }
                break;
            }
            data[index] = histori.toString();
            index++;
            histori.clear();
            System.out.print("\nMau main tebak lagi? (jawab 'YA' jika ingin)\n> ");
            response = input.nextLine();
        } while (response.equalsIgnoreCase("YA"));

        System.out.println("\nHasil permainan:");
        for (int i = 0; i < index; i+=2) {
            System.out.println("Pemain ke-" + ((i / 2) + 1));
            System.out.println("Nama: " + data[i]);
            System.out.println("Histori tebakan: " + data[i + 1]);
            System.out.println();
        }

        input.close();
    }
}

File: cpp/tebak.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    srand(time(0)); 
    int random = rand() % 101;
    int tebakan;
    int salah = 0;

    cout << "----------Permainan Tebak Angka----------" << endl;
    cout << "Tebaklah sebuah angka antara 0 hingga 100" << endl;

    while (true) {
        cout << "Masukkan tebakan Anda: ";
        cin >> tebakan;

        if (tebakan < 0 || tebakan > 100) {
            cout << "Tebakan diluar dari angka yang ditentukan. Silahkan coba lagi." << endl;
            continue;
        }

        if (tebakan > random) {
            cout << "Tebakan  Anda terlalu tinggi. Coba lagi." << endl;
            salah++;
            continue;
        } else if (tebakan < random) {
            cout << "Tebakan Anda terlalu rendah. Coba lagi." << endl;
            salah++;
            continue;
        } else if (tebakan == random) {
            cout << endl;
            cout << "--------------------------SELAMAT--------------------------" << endl;
            cout << "Anda berhasil menebak angka yang benar setelah " << salah << " percobaan" << endl;
            if (salah > 10) {
                cout << "Skill issue lu awokwkwk" << endl;
                cout << endl;
            } else {
                cout << endl;
            }
        }
        break;
    }

    return 0;
}